spec/use_cases/user_account_sync_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "rails_helper"
+
+RSpec.describe UserAccountSync do
+  let(:sqs_service) { instance_spy AwsService::SQS }
+  let(:use_case) { described_class.new sqs_service: sqs_service }
+
+  describe "#perform" do
+    let(:user) { build :user }
+    let(:expected_payload) {
+      {
+        id: user.id,
+        name: user.name,
+        email: user.email,
+        profile: user.profile.name,
+        sub_profile: user.sub_profile.try(:name),
+        gender: user.gender,
+        birthday: user.birthday.strftime("%Y-%m-%d"),
+        picture_url: user.picture.url,
+        password: user.encrypted_password,
+        cpf: user.cpf
+      }
+    }
+
+    let(:queue_name) { "user_sync_queue" }
+    before { ENV["USER_SYNC_QUEUE"] = queue_name }
```

Comment:
You are leaking this env to other tests. You can isolate the env change with something like this:

Add a spec support module:

```ruby
module WithEnvHelper
  # Sets env variables and executes the given block.
  #
  # The env is returned to the previous state after the block execution.
  #
  # @param kwargs [Hash<#to_s, #to_s>] the env variables
  def with_env(**kwargs, &block)
    old_envs = kwargs.each_with_object({}) do |(k, v), memo|
      env_name = to_env_name(k)

      memo[env_name] = ENV[env_name] if ENV.key?(env_name)
      ENV[env_name] = v.to_s
    end

    block.call
  ensure
    kwargs.each do |k, _v|
      env_name = to_env_name(k)

      if old_envs.key? env_name
        ENV[env_name] = old_envs[env_name]
      else
        ENV.delete env_name
      end
    end
  end

  private

  def to_env_name(key)
    key.to_s.upcase
  end
end
```

Now we can call code inside the block  say in a subject, or whenever you want.

```ruby
subject do
  with_env user_sync_queue: queue_name do
    # call my code which reads the env var
  end
end
```

Don't forget to add the module as a support lib.

```ruby
RSpec.configure do |config|
  config.include WithEnvHelper
...
```